### PR TITLE
Materialize site

### LINKED ISSUE #33: CSS Styling [8]
The site needs some CSS styling to look pretty. It'd also be good to minify our assets.

A good solution for both would be to use [Flask-Assets](https://flask-assets.readthedocs.io/en/latest/) to manage our assets.

### PR DESCRIPTION
**Pull request by subnomo**
Fixes #33 

This PR still needs work to improve usability, but I believe all that's left are "nice-to-haves," basic functionality should be there. If something is actually broken, please let me know.
## Changes in this PR.

- [X] Materialize existing pages
## Testing this PR.

1. Run `docker-compose build`
2. Start the container with `docker-compose up`
3. Visit http://localhost:5000, browse the site
### Expected Output.

You should see the materialized pages. This is an example from the `/times` page:
times page

@osuosl/devs

### CODE REVIEW

File: app/templates/create_time.html [hunk 1]
```diff
@@ -6,10 +6,30 @@
...
-        <p>{{ form.activities.label }} {{ form.activities }} {{ form.activities.description }}</p>
-        <p>{{ form.notes.label }} {{ form.notes }}</p>
-        <p>{{ form.issue_uri.label }} {{ form.issue_uri }} {{ form.issue_uri.description }}</p>
-        <p>{{ form.date_worked.label }} {{ form.date_worked }} {{ form.date_worked.description }}</p>
-        <p><input type="submit" value="Submit"></p>
+
+        {% if form.activities.choices %}
+            <p>{{ form.activities.label }} {{ form.activities }} {{ form.activities.description }}</p>
+        {% endif %}
+        {% if form.notes %}
```
Comment (aaroncohen73) [inline app/templates/create_time.html]: This seems like an unnecessary check to make

File: app/templates/layout.html [hunk 2]
```diff
@@ -2,23 +2,84 @@
...
-                <ul class="flashes">
-                {% for message in messages %}
-                    <li>{{ message | safe }}</li>
-                {% endfor %}
-                </ul>
-            {% endif %}
-        {% endwith %}
+        {# Sidebar navigation #}
+        <nav>
+            <ul id="slide-out" class="side-nav fixed">
```
Comment (aaroncohen73) [inline app/templates/layout.html:16]: The sidenav bar is missing links to create/view projects

File: app/__init__.py [hunk 3]
```diff
@@ -40,9 +41,23 @@
     print 'WARNING: The initialization vector has not been changed from the'
     print '         default. This is insecure, please generate a new one'
 
+assets = Environment(app)
+indexjs = Bundle('../js/index.js', filters='rjsmin', output='js/index.js')
```
Comment (aaroncohen73) [inline app/__init__.py]: When this is run after a `docker-compose up`, the files are owned by root. It's not like we'll need to edit these files by hand, but they should be renamed to `*.min.js` and/or put into a separate directory.

File: app/templates/view_users.html [hunk 4]
```diff
@@ -81,3 +62,20 @@
...
+{% block scripts %}
+    {% assets "js_forms" %}
+    <script type="text/javascript" src="{{ ASSET_URL }}"></script>
+    {% endassets %}
+
+<script>
+$(document).ready(function() {
+    var tableSelector = "#view-users table";
+    var perPage = 10;
+    var view_user_times_root = "{{ url_for('view_user_times') }}";
```
Comment (aaroncohen73) [inline app/templates/view_users.html]: `view_user_times` doesn't work

**Review by aaroncohen73 — CHANGES_REQUESTED** [review]
Overall, the website design looks pretty good. However, there are some things that aren't working. In addition to some of the notes I've left in the code,
- Table sorting doesn't work at all
- Multiselect inputs on forms appear as if an option is selected even if no option has been selected
- Form clearing doesn't work
- The pagination controls have disappeared entirely for some reason (There's an error in the browser console that might be helpful in figuring that one out).

**Review by aaroncohen73 — APPROVED** [review]

**Comment by aaroncohen73** [comment]
There seem to be just a few more things that need fixing before this is ready to merge. Fine-tuning the website design can come later.
- On some pages, not all of the sidebar links appear. For example, on the "Edit time" page the sidebar links to creating activities and projects don't appear. On some pages (The create/view projects pages and all of the edit and delete pages), the sidebar just shows the link to the login page.
- The edit project and view_user_times pages are broken.
- The following form inputs appear to be broken: The members/managers/spectators selection boxes on the "Create Project" page, all of the checkbox inputs

**Comment by subnomo** [comment]
@aaroncohen73 Thanks for the feedback, everything *should* be fixed now. Of course there are some usability tweaks to make, but that comes later.

**Comment by Kennric** [comment]
This looks pretty good to me - not perfect, but I'd rather get it in in and fix it than sit on it waiting for perfection.

**Pull request merged**